paint/data/stac_client.py

```diff
@@ -0,0 +1,560 @@
+            Child STAC catalog, collection, or item.
+        """
+        child = parent.get_child(child_id)
+        if child is None:
+            log.warning(
+                f"The child with ID {child_id} is not available, data for this child will not be downloaded."
+            )
+        return child
+
+    def download_file(self, url: str, file_name: Union[str, pathlib.Path]) -> None:
+        """
+        Download a file.
+
+        Parameters
+        ----------
+        url : str
+            URL of the file to download.
+        file_name : Union[str, pathlib.Path]
+            File name to be saved.
+        """
+        response = requests.get(url, stream=True)
+
+        # Open the file in binary write mode and download the content
+        with open(file_name, "wb") as file:
+            downloaded_length = 0
+            for data in response.iter_content(chunk_size=self.chunk_size):
+                file.write(data)
+                downloaded_length += len(data)
+
+    def process_heliostat_items(
+        self,
+        items: list[pystac.item.Item],
+        start_date: Optional[datetime],
+        end_date: Optional[datetime],
+        filtered_calibration_keys: Optional[list[str]],
+        collection_id: str,
+        heliostat_catalog_id: str,
+        save_folder: str,
+    ) -> None:
+        """
+        Process and download items, either with or without date filtering and calibration key filtering.
+
+        Parameters
+        ----------
+        items : list[pystac.item.Item]
+            List of items to be processed.
+        start_date : datetime, optional
+            Optional start date to filter the heliostat data. If no start date is provided, data for all time periods
+            is downloaded (Default is None).
+        end_date :  datetime, optional
+            Optional end date to filter the heliostat data. If no end date is provided, data for all time periods
+            is downloaded (Default is None).
+        filtered_calibration_keys : list[str]
+            List of keys to filter the calibration data. These keys must be one of:`raw_image, cropped_image,
+            flux_image, flux_centered_image, calibration_properties`. If no list is provided, all calibration data is
+            downloaded (Default is None).
+        collection_id : str
+            ID of the collection to download.
+        heliostat_catalog_id : str
+            The ID of the considered heliostat catalog.
+        save_folder : str
+            Name of the folder to save the collection items in.
+        """
+        for item in items:
+            item_time = item.properties["datetime"]
+            if start_date and end_date:
+                # If start and end dates are provided, filter based on them
+                if not (
+                    start_date
+                    <= datetime.strptime(item_time, mappings.TIME_FORMAT)
+                    <= end_date
+                ):
+                    continue
```

Comment:
Should we include a `log.debug` message stating that not data was found in the provided time span for this item?

Reply:
Done!